### ISSUE TITLE
Verify unused

### ISSUE THREAD

**Issue by som-snytt** [scala/bug#11631]
It would be nice if `-Xlint` also verified that an element annotated `unused` is in fact unused.

Currently it's just an exemption from the unused warning.

Probably it's not necessary to add an option, `-Xlint:used` for this case, although the LOL factor is tempting.

**Comment by dwijnand** [comment]
Agreed, there's no need for an option to disable verifying that the unused-parameter-suppressing annotation is on a parameter that is actually used.

**Comment by Jasper-M** [comment]
> Agreed, there's no need for an option to disable verifying that the unused-parameter-suppressing annotation is on a parameter that is actually used.

Alternatively I would suggest `-Xlint:confused`.

**Comment by som-snytt** [comment]
For social coding, `-Xlint:emoji` would hit github to check for :confused: or possibly (now that I notice it) :confounded: on the branch being built.

**Comment by lrytz** [comment]
Interestingly, this is "implemented" on 2.12 with [scala-collection-compat](https://github.com/scala/scala-collection-compat/blob/main/compat/src/main/scala-2.11_2.12/scala/annotation/unused.scala)

```scala
➜ sandbox scala repl -S 2.12 --dep org.scala-lang.modules::scala-collection-compat:2.13.0
Welcome to Scala 2.12.20 (OpenJDK 64-Bit Server VM, Java 21.0.1).
Type in expressions for evaluation. Or try :help.

scala> def f(@annotation.unused x: Int) = x
<console>:11: warning: value x is deprecated
       def f(@annotation.unused x: Int) = x
                                          ^
f: (x: Int)Int
```
